lib/transactional.pm

```diff
@@ -233,3 +238,70 @@ sub ensure_rollback_service_not_running {
         $output =~ '^(start|running)$' ? sleep 10 : last;
     }
 }
+
+=head2 enter_trup_shell
+
+  enter_trup_shell(global_options => $global_options, shell_options => $shell_options)
+
+Enter into transactional update shell by entering command on transactional server:
+transactional-update $global_options shell $shell_options. The two arguments for
+this subroutine, global_options and shell_options, are all text strings that are
+composed of space separated options for transactional-update and shell respectively.
+
+=cut
+
+sub enter_trup_shell {
+    my (%args) = @_;
+
+    $args{global_options} //= '';
+    $args{shell_options} //= '';
+    enter_cmd("transactional-update $args{global_options} shell $args{shell_options}; echo trup_shell-status-\$? > /dev/$serialdev");
+    save_screenshot;
+    wait_still_screen;
+    save_screenshot;
+}
+
+=head2 exit_trup_shell_and_reboot
+
+  exit_trup_shell_and_reboot()
+
+Quit transactional update shell by entering exit. Check if any changes that request
+reboot to take effect. This subroutine should be used together with enter_trup_shell.
+
+=cut
+
+sub exit_trup_shell_and_reboot {
+    enter_cmd("exit");
+    save_screenshot;
+    wait_serial('trup_shell-status-0') || croak("transactional-update shell did not finish");
+    save_screenshot;
+    assert_script_run("transactional-update status");
```

Comment:
Remove 
```
assert_script_run("transactional-update status");
```
or all these four lines ?
I think they are useful, because it makes sure user is really outside "trup shell". And assert_script_run("transactional-update status") is for user's information.

Reply:
> Remove
> 
> ```
> assert_script_run("transactional-update status");
> ```
> 
> or all these four lines ?

That + the `save_screenshot` around it.

> I think they are useful, because it makes sure user is really outside "trup shell". And assert_script_run("transactional-update status") is for user's information.

`transactional-update status` is experimental functionality which actually needs to be explicitly enabled before it works properly. It's not really meant to be used for this.